onnx/test/parser_test.py

```diff
@@ -119,6 +119,205 @@ def test_parse_function_with_attributes(self) -> None:
+                int_zero = Constant<value_int: int=0>()
+                int_four = Constant<value_int: int=4>()
+
+                constant_align_corners = Constant<value_int: int=@align_corners>()
+                constant_align_corners_equal_zero = Equal (constant_align_corners, int_zero) 
+                
+                size_ndim = Size (size)
+                condition_is_2d = Equal (size_ndim, int_four)
+
+                grid = If (condition_is_2d) <
+                    then_branch = g1 () => (float[N, H, W, 2] grid_2d_then) {
+                        minus_one = Constant <value = float {-1.0}>()
+                        zero = Constant <value = float {0.0}>()
+                        one = Constant <value = float {1.0}>()
+                        two = Constant <value = float {2.0}>()
+                        N, C, H, W = Split <num_outputs: int=4>(size)
+                        int_two_1d = Constant<value_ints=[2]>()
+                        int_four_1d = Constant<value_ints=[4]>()
+                        constant_H_W_shape = Slice(size, int_two_1d, int_four_1d) # [N, C, H, W] => [H, W]
+                        zero_H_by_W = ConstantOfShape (constant_H_W_shape)
+                        ones_H_by_W = Add (zero_H_by_W, one)
+
+                        H_float = CastLike (H, zero)
+                        W_float = CastLike (W, zero)
+                        grid_h_0, grid_w_0 = If (constant_align_corners_equal_zero) <
+                            then_branch = h1 () => (float[H] grid_h_then, float[W] grid_w_then) {
+                                step_h = Div (two, H_float)
+                                step_w = Div (two, W_float)
+                                step_h_half = Div (step_h, two)
+                                start_h = Add (minus_one, step_h_half)
+                                grid_h_then = Range (start_h, one, step_h)
+
+                                step_w_half = Div (step_w, two)
+                                start_w = Add (minus_one, step_w_half)
+                                grid_w_then = Range (start_w, one, step_w)
+                            },
+                            else_branch = h2 () => (float[H] grid_h_else, float[W] grid_w_else) {
+                                H_float_nimus_one = Sub (H_float, one)
+                                W_float_nimus_one = Sub (W_float, one)
+                                step_h = Div (two, H_float_nimus_one)
+                                step_w = Div (two, W_float_nimus_one)
+                                epsilon = Constant <value = float {1e-6}>()
+                                one_plus_epsilon = Add (one, epsilon)
+                                grid_h_else = Range (minus_one, one_plus_epsilon, step_h)
+                                grid_w_else = Range (minus_one, one_plus_epsilon, step_w)
+                            }
+                        >
+                        size_ones_H_by_W_transpose = Transpose (ones_H_by_W) # (3, 2)
```

Comment:
Doesn't this just produce `one_W_by_H` ? What does the `(3, 2)` mean?

Reply:
the comment (# (3, 2)) shall be removed. ones_W_by_H is a much better name.